ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -17800,6 +17800,31 @@ void HqlCppTranslator::doBuildAssignRegexFindReplace(BuildCtx & ctx, const CHqlB
     doBuildNewRegexFindReplace(ctx, &target, expr, NULL);
 }
 
+void HqlCppTranslator::doBuildExprRegexFindSet(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & bound)
+{
+    CHqlBoundExpr boundMatch;
+    if (ctx.getMatchExpr(expr, boundMatch))
+    {
+        bound.set(boundMatch);
+        return;
+    }
+
+    IHqlExpression * pattern = expr->queryChild(0);
+    IHqlExpression * search = expr->queryChild(1);
+    bool isUnicode = isUnicodeType(search->queryType());
+    IHqlExpression * compiled = doBuildRegexCompileInstance(ctx, pattern, isUnicode, !expr->hasAttribute(noCaseAtom));
+
+    // Because the search instance is created locally, the search parameter is always going to be valid
+    // as long as the find instance.  Only exception could be if call created a temporary class instance.
+    bool needToKeepSearchString = true;
```

Comment:
Why true?  If it false in the findreplace function, and it should really be false as far as I can see.

However it appears to be hiding two other bugs...so that's one reason.